kubernetes/samples/scripts/domain-lifecycle/startServer.sh

```diff
@@ -0,0 +1,252 @@
+  validateErrors=false
+
+  validateKubernetesCliAvailable
+  validateJqAvailable
+
+  # Validate that server name parameter is specified.
+  if [ -z "${serverName}" ]; then
+    validationError "Please specify a server name using '-s' parameter e.g. '-s managed-server1'."
+  fi
+
+  failIfValidationErrors
+}
+
+initialize
+
+# Get the domain in json format
+domainJson=$(${kubernetesCli} get domain ${domainUid} -n ${domainNamespace} -o json)
+if [ $? -ne 0 ]; then
+  printError "Unable to get domain resource. Please make sure 'domain_uid' and 'namespace' provided with '-d' and '-n' arguments are correct."
+  exit 1
+fi
+
+# Validate that specified server is either part of a cluster or is an independent managed server
+validateServerAndFindCluster "${domainUid}" "${domainNamespace}" "${serverName}" isValidServer clusterName
+if [ "${isValidServer}" != 'true' ]; then
+  printError "Server ${serverName} is not part of any cluster and it's not an independent managed server. Please make sure that server name specified is correct."
+  exit 1
+fi
+
+getClusterPolicy "${domainJson}" "${clusterName}" clusterPolicy
+if [ "${clusterPolicy}" == 'NEVER' ]; then
+  printError "Cannot start server '${serverName}', the server's parent cluster '.spec.clusters[?(clusterName=\"${clusterName}\"].serverStartPolicy' in the domain resource is set to 'NEVER'."
+  exit 1
+fi
+
+getDomainPolicy "${domainJson}" domainPolicy
+if [ "${domainPolicy}" == 'NEVER' ]; then
+  printError "Cannot start server '${serverName}', the .spec.serverStartPolicy in the domain resource is set to 'NEVER'."
+  exit 1
+fi
+
+getEffectivePolicy "${domainJson}" "${serverName}" "${clusterName}" effectivePolicy
+if [ -n "${clusterName}" ]; then
+  # Server is part of a cluster, check currently started servers
+  checkStartedServers "${domainJson}" "${serverName}" "${clusterName}" "${withReplicas}" "${withPolicy}" serverStarted
+  if [[ ${effectivePolicy} == "IF_NEEDED" && ${serverStarted} == "true" ]]; then
+    printInfo "No changes needed, exiting. The server should be already started or it's in the process of starting. The start policy for server ${serverName} is ${effectivePolicy} and server is chosen to be started based on current replica count."
+    exit 0
+  elif [[ "${effectivePolicy}" == "ALWAYS" && ${serverStarted} == "true" ]]; then
+    printInfo "No changes needed, exiting. The server should be already started or it's in the process of starting. The start policy for server ${serverName} is ${effectivePolicy}."
+    exit 0
+  fi
+else 
+  # Server is an independent managed server. 
+  if [[ "${effectivePolicy}" == "ALWAYS" || "${effectivePolicy}" == "IF_NEEDED" ]]; then
+    printInfo "No changes needed, exiting. The server should be already started or it's in the process of starting. The start policy for server ${serverName} is ${effectivePolicy}."
+    exit 0
+  fi
+fi
+
+getServerPolicy "${domainJson}" "${serverName}" managedServerPolicy
+targetPolicy="ALWAYS"
```

Comment:
the targetPolicy is never to anything other than "ALWAYS" - it doesn't need to be a variable